jina/types/request/data.py

```diff
@@ -22,14 +22,14 @@ class _DataContent:
         def __init__(self, content: 'jina_pb2.DataRequestProto.DataContentProto'):
             self._content = content
 
-        @property
+        @cached_property
```

Comment:
why we need this?

Reply:
it's needed to make sure request objects and DocumentArrays extracted from the request point to the same protobuf object. This helps perform reduce in-place without having to copy back to the request object.
Actually we noticed that sometimes Documents extracted from the request object aren't views to the same protobuf so we wanted to enforce the pattern in all cases

Reply:
then this will need to be reviewed with the new docArray